### PR TITLE
👷 Auto-commit pyfiction docstrings

### PR DESCRIPTION
**Pull request by simon1hofmann**
## Description

Auto-commit new pyfiction docstrings directly to your branch in a PR.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [ ] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by simon1hofmann** [comment]
@marcelwa Can we prioritize the run of `pyfiction-docstring-generator.yml` in some way? Otherwise, in the worst case, every CI will run again if `pyfiction-docstring-generator.yml` is the final CI to be run.

**Comment by marcelwa** [comment]
> @marcelwa Can we prioritize the run of `pyfiction-docstring-generator.yml` in some way? Otherwise, in the worst case, every CI will run again if `pyfiction-docstring-generator.yml` is the final CI to be run. 

We can make other workflows depend on this one. I.e. they are going to be stalled until this one finishes. Is that what we want?

**Comment by simon1hofmann** [comment]
> > @marcelwa Can we prioritize the run of `pyfiction-docstring-generator.yml` in some way? Otherwise, in the worst case, every CI will run again if `pyfiction-docstring-generator.yml` is the final CI to be run.
> 
> We can make other workflows depend on this one. I.e. they are going to be stalled until this one finishes. Is that what we want?

I think we don't have to necessarily stall the `Ubuntu`, `Windows` and `macOS` CI's, as updating `pybind_11_mkdoc_docstrings` does not change anything in the normal C++ build, so it should be enough to prevent rerunning them every time `pybind_11_mkdoc_docstrings` is updated?

Only the `pyfiction` CI's and the documentation build have to be stalled unitl the new docstrings have been generated.

**Comment by marcelwa** [comment]
> > > > > @marcelwa Can we prioritize the run of `pyfiction-docstring-generator.yml` in some way? Otherwise, in the worst case, every CI will run again if `pyfiction-docstring-generator.yml` is the final CI to be run.
> > > > 
> > > > 
> > > > We can make other workflows depend on this one. I.e. they are going to be stalled until this one finishes. Is that what we want?
> > > 
> > > 
> > > I think we don't have to necessarily stall the `Ubuntu`, `Windows` and `macOS` CI's, as updating `pybind_11_mkdoc_docstrings` does not change anything in the normal C++ build, so it should be enough to prevent rerunning them every time `pybind_11_mkdoc_docstrings` is updated?
> > > Only the `pyfiction` CI's and the documentation build have to be stalled unitl the new docstrings have been generated.
> > 
> > I don't know out of the top of my head how to do this. We have certain file patterns at the top of most CI configuration files that trigger these workflows on change. Maybe that's a good starting point for you to look into.
> 
> It should be possible to exclude specific paths, I've updated the corresponding workflows.

Many thanks! 🙏

File: .github/workflows/python-bindings.yml [hunk 1]
```diff
@@ -11,6 +11,11 @@ on:
       - '**/CMakeLists.txt'
       - 'libs/**'
       - '.github/workflows/python-bindings.yml'
+  workflow_run:
+    workflows: [ "pyfiction Docstring Generator" ]
+    branches: [ main ]
```
Comment (marcelwa) [inline .github/workflows/python-bindings.yml:16]: Does this mean, this run only depends on `pyfiction Docstring Generator` when run on `main`? Is that desired?

File: bindings/pyfiction/README.md [hunk 2]
```diff
@@ -81,16 +81,16 @@ file `include/documentation.hpp`, which acts as a wrapper around the aforementio
...
 
 #### Use the GitHub Action (preferred)
 
-Go to
-the [pyfiction-docstring-generator](https://github.com/cda-tum/fiction/actions/workflows/pyfiction-docstring-generator.yml)
-workflow on GitHub and run it. Upon completion, it will make the `pybind11_mkdoc_docstrings.hpp` file available as an
-artifact. Download the file and replace the existing one.
+Go to the [pyfiction-docstring-generator](https://github.com/cda-tum/fiction/actions/workflows/pyfiction-docstring-generator.yml)
+workflow on GitHub and run it. Upon completion, it will commit any changes in the `pybind11_mkdoc_docstrings.hpp` file to your branch.
+Alternatively, it will make the `pybind11_mkdoc_docstrings.hpp` file available as an artifact.
```
Comment (marcelwa) [inline bindings/pyfiction/README.md:93]: Is this still up to date? Feels like the workflow should no longer be manually triggerable now that everything is automated. 

Also, do these two paragraphs contain a lot of duplication now? They both talk about the GitHub Action but tell you one is preferred. Feels not right.

**Review by marcelwa — CHANGES_REQUESTED** [review]

**Comment by marcelwa** [comment]
@simon1hofmann would you mind resolving the conflict with the `pybind11_mkdoc_docstring.hpp` file now that wiring reduction was merged into `main`?

File: .github/workflows/python-bindings.yml (hunk 1 above)
Comment (simon1hofmann) [reply .github/workflows/python-bindings.yml]: Yes correct, I first thought it will depend on PR's to `main`, but that does not make any sense.

**Review by marcelwa — APPROVED** [review]

**Pull request merged**